tests/python/pants_test/backend/codegen/tasks/test_simple_codegen_task.py

```diff
@@ -0,0 +1,230 @@
+# coding=utf-8
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+from textwrap import dedent
+
+from pants.backend.codegen.register import build_file_aliases as register_codegen
+from pants.backend.codegen.tasks.simple_codegen_task import SimpleCodegenTask
+from pants.backend.core.register import build_file_aliases as register_core
+from pants.backend.jvm.targets.java_library import JavaLibrary
+from pants.backend.jvm.targets.jvm_target import JvmTarget
+from pants.base.build_file_aliases import BuildFileAliases
+from pants.base.payload import Payload
+from pants_test.tasks.task_test_base import TaskTestBase
+
+
+class SimpleCodegenTaskTest(TaskTestBase):
+  @classmethod
+  def task_type(cls):
+    return SimpleCodegenTask
```

Comment:
What I have done in the past to test base classes like this is to create a Dummy subclass and make it the target of the `task_type()`.  See `tests/python/pants_test/tasks/test_jvm_task.py`

I think this should give you an easy way to flexibly instantiate a concrete instance of SimpleCodegen without the complexity of `_create_subclass_task()` below.

```
class DummySimpleCodegenTask(SimpleCodegenTask):
  def execute(self):
    pass

def task_type(cls):
  return DummySimpleCodegenTask

...
  def test_isolated(self):
    context = self.context(options = { self.options_scope : {'strategy' : 'isolated'})
    isolated_task = self.create_task(context)
```